lib/valueflow.cpp

```diff
@@ -1551,27 +1589,6 @@ static bool valueFlowForward(Token * const               startToken,
                 }
             }
 
-            // bailout if address of var is taken..
```

Comment:
The reason is that the value flow does not set a value into the out parameter of a function.
```
void g(A & a);
void f() {
    ...
    g(b);
    ...
}
```
In the example above it prevents that the value flow sets the value of ```b``` in the function call even if the value of ```b``` is known. The reason for this is that ```b``` might be overwritten by ```g``` anyhow. I hope other parts of the code do not make use of the value to make further checks inside ```g```.
For the value flow of the move this is needed. Otherwise the following code gives a wrong warning that the moved variable a is accessed in the line of ```g(a)``` in the following code:
```
void g(A & a);
void f() {
    A a;
    A b = std::move(a);
    g(a);
    A c = a;
}
```
Note: This example if also part of the new test ```moveAndFunctionParameter2()```

If for some reason the original order is needed then it must be checked in ```CheckMove::checkAccess()``` if the moved variable is an (in-)out parameter of a function. This is possible, too.